test/modules/TestResultCoordinator/Reports/TestSummary.cs

```diff
@@ -1,24 +1,21 @@
 // Copyright (c) Microsoft. All rights reserved.
 namespace TestResultCoordinator.Reports
 {
-    using System;
+    using System.Collections.Generic;
     using System.Linq;
     using Microsoft.Azure.Devices.Edge.Util;
 
     class TestSummary
     {
-        internal TestSummary(string testBuildNumber, string deviceId, ITestResultReport[] testResultReports, string blobContainerUri)
+        internal TestSummary(Dictionary<string, string> testInfo, ITestResultReport[] testResultReports, string blobContainerUri)
         {
-            this.TestBuildNumber = Preconditions.CheckNonWhiteSpace(testBuildNumber, nameof(testBuildNumber));
-            this.DeviceId = Preconditions.CheckNonWhiteSpace(deviceId, nameof(deviceId));
+            this.TestInfo = Preconditions.CheckNotNull(testInfo, nameof(testInfo));
```

Comment:
nit: do we need to enforce not empty? your call?

Reply:
I think so, because this info will be relied upon for proper display in dashboard

Reply:
are you going to check dictionary is not empty?

Reply:
I'm doing the check in ModuleUtil